compiler/rustc_codegen_ssa/src/back/metadata.rs

```diff
@@ -60,16 +60,24 @@ impl MetadataLoader for DefaultMetadataLoader {
                     let data = entry
                         .data(data)
                         .map_err(|e| format!("failed to parse rlib '{}': {}", path.display(), e))?;
-                    return search_for_section(path, data, ".rmeta");
+                    if target.is_like_aix {
```

Comment:
Though xcoff is bound to AIX, I think checking the format is more precise.

Reply:
We don't have format info in this function. While according to your comments, I changed such checks in `fn create_wrapper_file()` and `fn create_compressed_metadata_file()`. Thanks.